js/btcmarkets.js

```diff
@@ -85,6 +86,104 @@ module.exports = class btcmarkets extends Exchange {
         });
     }
 
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'limit': limit,
+            'since': since,
+        };
+        const response = await this.privateGetFundtransferHistory (this.extend (request, params));
+        if (!response['success'])
+            throw new ExchangeError (this.id + ' fetch transactions failed ' + this.json (response));
+        const results = [];
+        for (let i = 0; i < response['fundTransfers'].length; i++) {
```

Comment:
We need to detach the parsing section from the `fetchMethod`, namely, call the base  `parseTransactions` method here and define `parseTransaction` and `parseTransactionStatus`, as shown in the existing exchanges (search for `parseTransactions`, `parseTransaction`, `parseTransactionStatus` in the code).

Reply:
It's the same thing I did with `therock` as I mentioned here: https://github.com/ccxt/ccxt/pull/5231#discussion_r290308361 – i could merge these PRs a lot faster, if you could carry the edits I'm making in the previously merged PRs - to the PRs that follow next – that would really speed it up the process. Ideally, I'd merge them without editing at all, if you can consistently reproduce the architecture taking, say, the certified exchanges as examples.

Reply:
I thought the point you were making about the rock was that I should have implemented fetchLedger instead of fetchTransactions?

Reply:
... but regarding the missing method, my bad.  Sorry, I submitted this prematurely.

Reply:
When implementing fetchDeposts and fetchWithdrawals separately, how do you call a unified parseTransaction if the type isn't in the data?

Reply:
> I thought the point you were making about the rock was that I should have implemented fetchLedger instead of fetchTransactions?

Yes, primarily, it was about `fetchTransactions` and `fetchLedger`, but I also added a proper error handler and a more or less complete exceptions map to it ) In other words, when I meet my own old code that was implemented in a hurry, i'm rewriting it for the internal standards that have developed further since then. In your code there are checks for response['result'] everywhere, therefore I'd suggest to move those check into a clean and shiny `handleErrors` along the the other work on this file )

Reply:
> When implementing fetchDeposts and fetchWithdrawals separately, how do you call a unified parseTransaction if the type isn't in the data?

We're looking for other strong differences between deposits and withdrawals to see if we can deduce the type from a combination and/or presence of other fields. If there's no way to determine the type of the transaction from the transaction structure itself, then we're not parsing the type in `parseTransactions` – we've added an extra `params` argument to `parseTransactions` that would extend all the parsed structures with a templated object:

```JavaScript
async fetchDeposits (...) {
    // ...
    return this.parseTransactions (response, currency, since, limit, {
        'type': 'deposit',
    });
}

async fetchWithdarwals (...) {
    // ...
    return this.parseTransactions (response, currency, since, limit, {
        'type': 'withdrawal',
    });
}
```

Reply:
With regard to the error handling, I had copied from another method.  My bad.  Looks like there is already some error handling in the correct place so I removed it.

Reply:
> Looks like there is already some error handling in the correct place so I removed it.

Yep, we might need to extend the exceptions map, if you've encountered a new type of exception: https://github.com/ccxt/ccxt/pull/5249/files#diff-f372fe18a8cb538a8274af517bb63562R83